add kafka brokers fail to start due to full disk doc

Jacky-hdi opened the pull request.

Jacky-hdi committed:
Create kafka-troubleshoot-full-disk

add kafka brokers fail to start due to a full disk

Jacky-hdi committed:
Merge branch 'MicrosoftDocs:master' into master

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+---
+title: Broker fails to start due to a full disk in Azure HDInsight
+description: Kafka broker process cannot start since it fails due to disk full error. Any config change will not take into effect since broker process is not starting.
```
```suggestion
description: Troubleshooting steps for Apache Kafka broker process that does not cannot start due to disk full error. 
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+
+## Issue
+
+Few of the data disks which are used by kafka brokers is full. Kafka broker process cannot start since it fails due to disk full error. Any config change will not take into effect since broker process is not starting.
```
Grammar needs fixing. I don't understand the first sentence.

```suggestion
The data disks used by Apache Kafka brokers in Azure HDInsight clusters can fill up. When that happens, the Apache Kafka broker process cannot start since it fails due to the disk full error. If you have made any recent configuration changes, those changes do not take into effect since the Kafka broker process doesn't start.
```

Open ends needing to be explained:
1. What's the exact disk full error text. Where do you see the error?
2. What disks exactly? Explain how one can identify this problem.
3. When does this problem happen - high stress, aged cluster, busy processing? Are there reasons why the used disk space grows?

Jacky-hdi replied:
Thanks for your suggestion.
1. As mentioned in the cause section, customer would see general disk alert and kafka broker process connection alert in Ambari UI. Then they tried to restart broker but failed. From customer's standpoint, they cannot produce/consume messages or create/delete topics due to unhealthy brokers
2. It depends on the log or message directory configuration. It usually happens on kafka data disks by default. We can identify through the ambari data disk alerts and kafka broker process alert. At the same time, we can use the 'find' command in the below to identify the biggest directory.
3. Inadequate topic partititons can cause big partition which occupies the space in the worker node. Also the load or amount of data ingestion increases would also lead to this.

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+
+## Cause
+
+- It's impossible to increase the number of disks or increase the disk size once we have created the kafka cluster.
```
```suggestion
- You cannot increase the number of disks or increase the disk size once we have created the Kafka cluster.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+## Cause
+
+- It's impossible to increase the number of disks or increase the disk size once we have created the kafka cluster.
+- Usually, disk alerts are configured on ambari UI. Whenever the usage of disk increases from 60%, it gives an alert that we need to either scale out or reduce the amount of logs present in kafka cluster.
```
```suggestion
- Usually, disk alerts are configured in the Apache Ambari UI. Whenever the usage of disk increases from 60%, it gives an alert that we need to either scale out or reduce the amount of logs present in Kafka cluster.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+
+- It's impossible to increase the number of disks or increase the disk size once we have created the kafka cluster.
+- Usually, disk alerts are configured on ambari UI. Whenever the usage of disk increases from 60%, it gives an alert that we need to either scale out or reduce the amount of logs present in kafka cluster.
+- This issue will happen only if you ignore the disk alerts and didn't scale out properly. 
```
Sounds accusatory. Can we soften the message please?
```suggestion
- This issue will happen only if you ignore the disk alerts. You could scale out the cluster to make more space available to respond to the disk space alerts.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+
+## Mitigation
+
+1. Check `server.properties` filesTo find the retention time for all the topics. Sometimes log retention policy is set at topic level. To find retention time configured at topic level, please run the following command:
```
```suggestion
1. Check the `server.properties` files to find the retention time for every topic. Sometimes the log retention policy is set at topic level. To find retention time configured at topic level, run the following command:
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
````diff
@@ -0,0 +1,83 @@
+
+```
+/usr/hdp/current/kafka-broker/bin/kafka-topics.sh --describe --zookeeper <zookeeper-list> 
+```
````
Label the code blocks as Bash commands for all I think:

```suggestion
```bash
/usr/hdp/current/kafka-broker/bin/kafka-topics.sh --describe --zookeeper <zookeeper-list> 
```
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
````diff
@@ -0,0 +1,83 @@
+```
+
+
+2. Once we have that information, we need to check for heavy partitions which occupy the maximum space.
````
```suggestion
2. Once you have that information, check which heavy partitions occupy the maximum disk space.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
````diff
@@ -0,0 +1,83 @@
+```
+# Command to sort the directories based on size:
+du -hs * | sort -rh | head -5 
+```
````
```suggestion
```bash
# Command to sort the directories based on size:
du -hs * | sort -rh | head -5 
```
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+
+* Connect with [@AzureSupport](https://twitter.com/azuresupport) - the official Microsoft Azure account for improving customer experience. Connecting the Azure community to the right resources: answers, support, and experts.
+
+* If you need more help, you can submit a support request from the [Azure portal](https://portal.azure.com/?#blade/Microsoft_Azure_Support/HelpAndSupportBlade/). Select **Support** from the menu bar or open the **Help + support** hub. For more detailed information, review [How to create an Azure support request](../../azure-portal/supportability/how-to-create-azure-support-request.md). Access to  ...
```
These lines are already part of a templated include file: /hdinsight/includes/hdinsight-troubleshooting-next-steps.md

Just embed the include and remove the text in the Next steps.

```suggestion
[!INCLUDE [troubleshooting next steps](../includes/hdinsight-troubleshooting-next-steps.md)]
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+- If the partitions is too big, increase the number of partitions for heavy topics
+
+> [!NOTE] 
+> But increasing the partitions in an existing topic doesn't reorganize the data in the topic. You have to manually copy data from a old low partition topic to a new higher partition topic in the first. 
```
Indent the note
```suggestion
   > [!NOTE] 
   > But increasing the partitions in an existing topic doesn't reorganize the data in the topic. You have to manually copy data from a old low partition topic to a new higher partition topic in the first. 
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
```diff
@@ -0,0 +1,83 @@
+Even though you can decrease the retention time, it will not be scalable if you want to add more topics in the cluster or if the load or amount of data ingested increases.
+To avoid the above the scenario these are the following options:
+
+- If the partitions is too big, increase the number of partitions for heavy topics
```
Grammar check 
```suggestion
- If the partitions are too big, increase the number of partitions for the heaviest topics.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
````diff
@@ -0,0 +1,83 @@
+# Command to move files older than 4 days:
+find . -maxdepth 1 -mtime +4 -type f -exec mv "{}" /destination/directory/ \;
+```
+4. When there is some free space available, you can restart the brokers with new retention time. This will clean the older logs and free up some space on disk.
````
```suggestion
4. When there is some free space available, you can restart the brokers with the new retention time configuration. This will clean the older logs and free up some space on disk.
```

JasonWHowell commented on articles/hdinsight/kafka/kafka-troubleshoot-full-disk:
````diff
@@ -0,0 +1,83 @@
+```
+# Command to move files older than 4 days:
+find . -maxdepth 1 -mtime +4 -type f -exec mv "{}" /destination/directory/ \;
+```
````
Indent all the lines under a numbered bullet.
```suggestion

   ```
   # Command to find files older than 4 days:
   find . -maxdepth 1 -mtime +4 -type f -exec ls -l "{}"  \; 
   ```

   > [!WARNING] 
   > Make sure **NOT** to remove/move `_meta.properties_` file in "/data_disk_*/kafka-logs" directory. 
   > Deleting `_meta.properties_` file causes broker ids to get updated when Kafka restarts. Topics which were created with old broker ids become under replicated or offline so the cluster goes in a bad state.

   ```
   # Command to move files older than 4 days:
   find . -maxdepth 1 -mtime +4 -type f -exec mv "{}" /destination/directory/ \;
   ```
```

JasonWHowell reviewed (changes requested):
Thanks for the troubleshooting article. 

There are too many problems to keep this page in the public repo. Please use azure-docs-pr for this contribution.

There needs to be a TOC.yml entry for the new page to make it show up in the table of contents.

A few themes on the issues
- The text needs to use official trademark branding for Apache * products such as Kafka and Ambari. Capitalized names, and use Apache when appropriate.
- Indentation throughout to align sections to the numbered sequences.
- Grammar and spelling check with Acrolinx

Some inline suggestions to fix a few of those issues:

Jacky-hdi committed:
Apply suggestions from code review

Troubleshooting steps for Apache Kafka broker process that cannot start due to disk full error.

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

This issue will happen if you ignore the disk alerts. You could scale out the cluster to make more space available to respond to the disk space alerts. Sometimes inadequate topic partitions which leads to big partition which occupies the most space in the worker node

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>

Jacky-hdi committed:
Update articles/hdinsight/kafka/kafka-troubleshoot-full-disk

The data disks used by Apache Kafka brokers in Azure HDInsight clusters can fill up. When that happens, the Apache Kafka broker process can be unhealthy. You cannot produce/consume messages or create/delete topics. Unfortunately, you also cannot restart. If you have made any recent configuration changes, those changes do not take into effect since the Kafka broker process doesn't start.

Co-authored-by: Jason Howell <5067358+JasonWHowell@users.noreply.github.com>
